src/main/java/gregtech/common/metatileentities/primitive/MetaTileEntityCharcoalPileIgniter.java

```diff
@@ -0,0 +1,511 @@
+        return data;
+    }
+
+    @Override
+    public void readFromNBT(NBTTagCompound data) {
+        super.readFromNBT(data);
+        this.lDist = data.hasKey("lDist") ? data.getInteger("lDist") : this.lDist;
+        this.rDist = data.hasKey("rDist") ? data.getInteger("rDist") : this.rDist;
+        this.hDist = data.hasKey("hDist") ? data.getInteger("hDist") : this.hDist;
+        this.progressTime = data.getInteger("progressTime");
+        this.maxProgress = data.getInteger("maxProgress");
+        this.isActive = data.getBoolean("isActive");
+    }
+
+    @Override
+    public void writeInitialSyncData(PacketBuffer buf) {
+        super.writeInitialSyncData(buf);
+        buf.writeInt(this.lDist);
+        buf.writeInt(this.rDist);
+        buf.writeInt(this.hDist);
+        buf.writeInt(this.progressTime);
+        buf.writeInt(this.maxProgress);
+        buf.writeBoolean(this.isActive);
+    }
+
+    @Override
+    public void receiveInitialSyncData(PacketBuffer buf) {
+        super.receiveInitialSyncData(buf);
+        this.lDist = buf.readInt();
+        this.rDist = buf.readInt();
+        this.hDist = buf.readInt();
+        this.progressTime = buf.readInt();
+        this.maxProgress = buf.readInt();
+        this.isActive = buf.readBoolean();
+    }
+
+    @Override
+    public void receiveCustomData(int dataId, PacketBuffer buf) {
+        super.receiveCustomData(dataId, buf);
+        if (dataId == GregtechDataCodes.UPDATE_STRUCTURE_SIZE) {
+            this.lDist = buf.readInt();
+            this.rDist = buf.readInt();
+            this.hDist = buf.readInt();
+            this.reinitializeStructurePattern();
+        } else if (dataId == GregtechDataCodes.WORKABLE_ACTIVE) {
+            this.isActive = buf.readBoolean();
+            scheduleRenderUpdate();
+        }
+    }
+
+    /**
+     * Add a block to the valid Charcoal Pile valid wall/roof blocks
+     * @param block the block to add
+     */
+    @SuppressWarnings("unused")
+    public static void addWallBlock(@Nonnull Block block) {
+        WALL_BLOCKS.add(block);
+    }
+
+    @ZenMethod("addWallBlock")
+    @Optional.Method(modid = GTValues.MODID_CT)
+    @SuppressWarnings("unused")
+    public static void addWallBlockCT(@Nonnull IBlock block) {
+        WALL_BLOCKS.add(CraftTweakerMC.getBlock(block));
+    }
+
+    @Override
+    public boolean isWorkingEnabled() {
+        return true;
+    }
+
+    @Override
+    public void setWorkingEnabled(boolean isActivationAllowed) {
+
+    }
+
+    @Override
+    public int getProgress() {
+        return progressTime;
+    }
+
+    @Override
+    public int getMaxProgress() {
+        return maxProgress;
+    }
+
+    @Override
+    public boolean isActive() {
+        return this.isActive;
+    }
+
+    @Override
+    public <T> T getCapability(Capability<T> capability, EnumFacing side) {
+        if (capability == GregtechTileCapabilities.CAPABILITY_CONTROLLABLE) {
```

Comment:
Does this structure even care about being controllable? I would think that it does not, since it makes no sense to be able to pause the structure mid recipe.